src/main/java/sparksoniq/jsoniq/runtime/iterator/primary/ArrayRuntimeIterator.java

```diff
@@ -32,23 +32,27 @@
 
 public class ArrayRuntimeIterator extends LocalRuntimeIterator {
 
-
     private static final long serialVersionUID = 1L;
 
     public ArrayRuntimeIterator(RuntimeIterator arrayItems, IteratorMetadata iteratorMetadata) {
-        super(new ArrayList<>(), iteratorMetadata);
-        if (arrayItems != null)
+        super(null, iteratorMetadata);
+        if (arrayItems != null) {
             this._children.add(arrayItems);
+        }
     }
 
     @Override
     public Item next() {
         if (this._hasNext) {
-            List<Item> result = this.runChildrenIterators(this._currentDynamicContextForLocalExecution);
+            List<Item> result = new ArrayList<>();
+            if (!this._children.isEmpty()) {
+                result.addAll(this._children.get(0).materialize(_currentDynamicContextForLocalExecution));
```

Comment:
Don't we need to iterator over the children here to keep the same behavior?

This may also be achieved with a materializeAllChildrenIteratorsIntoConcatenatedSequence() method (or a name like this).

Except of, of course, we know for sure that there is only child iterator anyway...

Reply:
Checked offline: only one child